Source/Blazorise/Components/ModalProvider/ModalProvider.razor.cs

```diff
@@ -34,37 +35,110 @@ public partial class ModalProvider
 
         #region Methods
 
+        ///inheritdoc
         protected override Task OnInitializedAsync()
         {
             ModalService.SetModalProvider( this );
             return base.OnInitializedAsync();
         }
-        public Task Show( RenderFragment childContent )
+
+
+        internal Task Show( RenderFragment childContent )
         {
             this.childContent = childContent;
             return modalRef.Show();
         }
 
-        public Task Hide()
+
+        internal Task Hide()
             => modalRef.Hide();
 
         #endregion
 
         #region Properties
 
+        ///inheritdoc
         [Inject] public IModalService ModalService { get; set; }
 
         #endregion
     }
 
     /// <summary>
-    /// Together with ModalProvider, handles the instantion of modals with custom content.
+    /// Sets the options for Modal Provider
+    /// </summary>
+    public class ModalProviderOptions
+    {
+        /// <summary>
+        /// Uses the modal standard structure, by setting this to true you are only in charge of providing the custom content.
+        /// Defaults to true.
+        /// </summary>
+        public bool UseModalStructure { get; set; } = true;
+
+        /// <summary>
+        /// Gets or Sets the modal's title.
+        /// </summary>
+        public string Title { get; set; }
+
+    }
+
+    /// <summary>
+    /// Provide the parameters to be passed onto your custom content to be rendered under the modal.
+    /// </summary>
+    public class ModalComponentParameter
```

Comment:
Name if ModalProviderParameter? Also, make it a struct instead.

Reply:
Maybe ModalProviderComponentParameter or content? The idea is that it represents the user custom component parameters. I guess if we don't need additional behaviour or helper methods we can make the dictionary.

The advantage is that if in the future we need to add behaviour for some reason we don't need to change the api.

Structure change seems fine. I always forget nowadays haha